src/server/controllers/user.js

```diff
@@ -0,0 +1,70 @@
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
+const secret = process.env.JWT_SECRET
+const jwt = require('jsonwebtoken')
+const bcrypt = require('bcrypt')
+const saltRounds = 10
+
+const createUser = async(req, res) => {
+    const {username, password} = req.body
+
+    if (username && password) {
+        bcrypt.genSalt(saltRounds, function (err, salt) {
+            bcrypt.hash(password, salt, async function (err, hash) {
+                try {
+
+                    const user = await prisma.user.create({
+                        data: {
+                            username,
+                            password: hash
+                        }
+                    })
+    
+                    res.status(201).json({user})
+                } catch (e) {
+                    const error = "The user already exists"
+                    
+                    res.status(409).send({error})
```

Comment:
I like the way you've handled verifying the right information has been provided by the user before encrypting and adding to the db!